src/relay/backend/contrib/cmsisnn/relay_to_tir.cc

```diff
@@ -0,0 +1,147 @@
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <tvm/relay/expr_functor.h>
+#include <tvm/tir/builtin.h>
+#include <tvm/tir/expr.h>
+#include <tvm/tir/function.h>
+#include <tvm/tir/op.h>
+#include <tvm/tir/stmt_functor.h>
+
+#include "../../../qnn/utils.h"
+
+namespace tvm {
+namespace relay {
+
+class RelayToTIR : public MixedModeVisitor {
+ public:
+  explicit RelayToTIR(String func_name) : func_name_(func_name) {}
+
+ private:
+  bool is_quant_softmax(const CallNode* call) {
```

Comment:
I dont think this check is valuable.

We could just check whether its a call to the composite function : cmsisnn.qnn_softmax